msgpack/msgpack.go

```diff
@@ -0,0 +1,137 @@
+package mc_msgpack
+
+import (
+	"io"
+	"reflect"
+
+	mc "github.com/multiformats/go-multicodec"
+	gocodec "github.com/ugorji/go/codec"
+)
+
+var HeaderPath string
+var Header []byte
+
+// Handler options. See go/codec docs for their meanings.
+var (
+	// MsgpackHandle
+	RawToString bool
+	WriteExt    bool
+
+	// BasicHandle
+	TypeInfos *gocodec.TypeInfos
+
+	// Encode options
+	StructToArray       bool
+	Canonical           bool
+	CheckCircularRef    bool
+	RecursiveEmptyCheck bool
+	Raw                 bool
+	AsSymbols           gocodec.AsSymbolFlag
+
+	// Decode options
+	MapType              reflect.Type
```

Comment:
I see that they are used as defaults. Perhaps have a `DefaultEncoder` that you can just copy when creating a new decoder/encoder